backend/plonk/bn254/solidity.go

```diff
@@ -603,30 +611,36 @@ contract PlonkVerifier {
       {{ if (gt (len .CommitmentConstraintIndexes) 0 )}}
       // compute the contribution of the public inputs whose indices are in commitment_indices,
       // and whose value is hash_fr of the corresponding commitme
-      uint256[] memory commitment_indices = new uint256[](vk_nb_commitments_commit_api);
+      uint256[] memory commitment_indices;
+      commitment_indices = new uint256[](vk_nb_commitments_commit_api);
       load_vk_commitments_indices_commit_api(commitment_indices);
 
       uint256[] memory wire_committed_commitments;
-      wire_committed_commitments = new uint256[](2*vk_nb_commitments_commit_api);
+      unchecked {
+        wire_committed_commitments = new uint256[](2 * vk_nb_commitments_commit_api);
+      }
 
       load_wire_commitments_commit_api(wire_committed_commitments, proof);
 
-      for (uint256 i=0; i<vk_nb_commitments_commit_api; i++){
-          
-          uint256 hash_res = Utils.hash_fr(wire_committed_commitments[2*i], wire_committed_commitments[2*i+1]);
-          uint256 a = compute_ith_lagrange_at_z(zeta, zeta_power_n_minus_one, commitment_indices[i]+public_inputs.length);
-          assembly {
-            a := mulmod(hash_res, a, r_mod)
-            pi := addmod(pi, a, r_mod)
-          }
+
+      for (uint256 i; i < vk_nb_commitments_commit_api; ) {
+        uint256 hash_res = Utils.hash_fr(wire_committed_commitments[2 * i], wire_committed_commitments[2 * i + 1]);
```

Comment:
this can also be replaced with 
```
      uint256 hash_fr_x;
      uint256 hash_fr_y;
      unchecked{
        hash_fr_x = wire_committed_commitments[2 * i];
        hash_fr_y = wire_committed_commitments[2 * i + 1];
      }

      uint256 hash_res = Utils.hash_fr(hash_fr_x, hash_fr_y);
```

Another 627 gas saved

Reply:
oh nice catch^^ I would never have found that actually, even when you assign values there is a check??

Reply:
ah right